### PR TITLE
refactor: Extract `SlideOver` component to `@faststore/ui`

### PR DESCRIPTION
**Pull request by filipewl**
## What's the purpose of this pull request?

Extract the `SlideOver` component to the `@faststore/ui` package. The component has been moved 

## How to test it?

Check that the component is still functional in the following previews:
- https://nextjs-store-storybook-git-refactor-fs-360-slide-over-faststore.vercel.app/?path=/docs/organisms-slideover--default-story
- https://preview-225--nextjs.preview.vtex.app/ (e.g. open the mini cart).

## References

- https://github.com/vtex/faststore/pull/1447

## Checklist

**Dependencies**
- [ ] Committed the `yarn.lock` and `bun.lockb` file when there were changes to the packages

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "@faststore/api": "^1.11.7",
     "@faststore/graphql-utils": "^1.10.34",
     "@faststore/sdk": "^1.10.34",
-    "@faststore/ui": "^1.11.2",
+    "@faststore/ui": "https://pkg.csb.dev/vtex/faststore/commit/a83d7380/@faststore/ui",
```
Comment (hellofanny) [inline package.json:37]: 🎗️ Remind to upgrade faststore/ui version when available

File: yarn.lock [hunk 2]
```diff
@@ -1479,10 +1479,9 @@
   dependencies:
     idb-keyval "^5.1.3"
 
-"@faststore/ui@^1.11.2":
+"@faststore/ui@https://pkg.csb.dev/vtex/faststore/commit/a83d7380/@faststore/ui":
```
Comment (hellofanny) [inline yarn.lock:1482]: 🎗️ Remind to upgrade faststore/ui version when available

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 3]
```diff
@@ -1,57 +1,16 @@
-import { Modal as UIModal } from '@faststore/ui'
-import type { HTMLAttributes, ReactNode } from 'react'
+import { SlideOver as SlideOverUI } from '@faststore/ui'
```
Comment (hellofanny) [inline src/components/ui/SlideOver/SlideOver.tsx:1]: In this case, is there any specific reason for using `UI` in the end?
We are usually using UI in the beginning. Can we keep this pattern? 👀

Comment (hellofanny) [inline src/components/ui/SlideOver/SlideOver.tsx:1]:
```suggestion
import { SlideOver as UISlideOver } from '@faststore/ui'
```

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 4]
```diff
@@ -1,57 +1,16 @@
...
+import type { SlideOverProps as SlideOverUIProps } from '@faststore/ui'
```
Comment (hellofanny) [inline src/components/ui/SlideOver/SlideOver.tsx:2]:
```suggestion
import type { SlideOverProps as UISlideOverProps } from '@faststore/ui'
```

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 5]
```diff
@@ -1,57 +1,16 @@
...
-  size = 'full',
-  fade = 'out',
-  children,
-  className,
-  ...otherProps
-}: SlideOverProps) {
+function SlideOver({ className, children, ...otherProps }: SlideOverUIProps) {
   return (
-    <UIModal
+    <SlideOverUI
```
Comment (hellofanny) [inline src/components/ui/SlideOver/SlideOver.tsx:8]:
```suggestion
    <UISlideOver
```

File: src/components/ui/SlideOver/SlideOver.tsx [hunk 6]
```diff
@@ -1,57 +1,16 @@
...
-      isOpen={isOpen}
-      data-fs-slide-over
-      data-fs-slide-over-direction={direction}
-      data-fs-slide-over-size={size}
-      data-fs-slide-over-state={fade}
       {...otherProps}
     >
       {children}
-    </UIModal>
+    </SlideOverUI>
```
Comment (hellofanny) [inline src/components/ui/SlideOver/SlideOver.tsx:13]:
```suggestion
    </UISlideOver>
```

**Comment by filipewl** [comment]
Closing this as we have other directions now.

**Pull request closed without merging**